### PR TITLE
Add hints to commit/push errors

### PR DESCRIPTION
**Pull request by tbantle22**

### CODE REVIEW

File: go/cmd/dolt/commands/push.go [hunk 1]
```diff
@@ -392,6 +392,11 @@ func pushToRemoteBranch(ctx context.Context, dEnv *env.DoltEnv, mode ref.RefUpda
 				cli.Println("hint: 'dolt pull ...') before pushing again.")
 				return errhand.BuildDError("").Build()
 			} else {
+				errString := err.Error()
+				if errString == "rpc error: code = PermissionDenied desc = permission denied" {
```
Comment (tbantle22) [inline go/cmd/dolt/commands/push.go:396]: Is there a better way to match this error?

**Comment by timsehn** [comment]
This is a good change. @andrew-wm-arthur do you mind reviewing since @bheni is out?

File: go/cmd/dolt/commands/push.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply go/cmd/dolt/commands/push.go:396]: You should be able to match RPC errors with this method: https://godoc.org/google.golang.org/grpc/status#FromError

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM, one suggestion

File: go/cmd/dolt/commands/push.go (hunk 1 above)
Comment (tbantle22) [reply go/cmd/dolt/commands/push.go]: Fixed https://github.com/dolthub/dolt/pull/1047/commits/6089d7e15d5fe4b02a4dc13630289baee7f937b0

**Pull request merged**